scripts/esbuild-targets.mjs

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env node
+
+import browserslist from 'browserslist';
+
+/**
+ * @see https://esbuild.github.io/api/#target
+ */
+const SUPPORTED_ESBUILD_TARGETS = new Set([
+  'chrome',
+  'deno',
+  'edge',
+  'firefox',
+  'hermes',
+  'ie',
+  'ios',
+  'node',
+  'opera',
+  'rhino',
+  'safari',
+]);
+
+process.stdout.write(
+  browserslist()
+    // Convert name and version range to parts
+    // ["chrome 119" ...] ➡️ [["chrome", "119"] ...]
+    .map((browser) => browser.split(' '))
+    // Use the lower version of a range
+    // [["ios_saf", "16.6-16.7"] ...] ➡️ [["ios_saf", "16.6"] ...]
+    .map(([name, versions]) => [name, versions.split('-')[0]])
+    // Exclude unsupported targets
+    // [["samsung", "23"] ...]  ➡️ [...]
+    .filter(([name]) => SUPPORTED_ESBUILD_TARGETS.has(name))
+    // Use the lowest version when multiple versions of the same browser are supported
+    // [["chrome", "119"], ["chrome", "118"], ...] ➡️ [["chrome", "118"] ...]
+    .filter(([name, version], _index, array) =>
+      array.every(([otherName, otherVersion]) => otherName !== name || version <= otherVersion),
+    )
+    // Convert back to ESBuild string target format
+    // [["chrome", "118"], ...] ➡️ ["chrome118" ...]
+    .map(([name, version]) => [name, version].join(''))
+    // Convert to comma separated list
+    // ["chrome118", "safari15.6" ...] ➡️ "chrome118,safari15.6"
+    .join(),
```

Comment:
the comments are great! if this was changed often we might want to run unit tests for it?

Reply:
Yeah, I suspect that'd be helpful, though there's not a great pattern for testing the logic of a file like this. Ideally we'd move the transformation code into its own package, and I do think component-specific packages (like USWDS's [packages](https://github.com/uswds/uswds/tree/develop/packages) pseudo-monorepo) makes sense for the design system.

Another option could be to use an existing package that does this already, like [`browserslist-to-esbuild`](https://github.com/marcofugaro/browserslist-to-esbuild) or [`esbuild-plugin-browserslist`](https://github.com/nihalgonsalves/esbuild-plugin-browserslist), though AFAICT neither of those can be used via the CLI.